provider/backingchain/blockcommand_base.py

```diff
@@ -99,14 +100,21 @@ def prepare_snapshot(self, start_num=0, snap_num=3,
                 path = snap_path
             if os.path.exists(path) and "reuse" not in option and clean_snap_file:
                 libvirt.delete_local_disk('file', path)
+            snap_name = 'snap%d' % i
             snap_option = "%s %s --diskspec %s,file=%s%s" % \
-                          ('snap%d' % i, option, self.new_dev, path, extra)
+                          (snap_name, option, self.new_dev, path, extra)
 
             virsh.snapshot_create_as(self.vm.name, snap_option,
                                      ignore_status=False,
                                      debug=True)
             self.snap_path_list.append(path)
-            self.snap_name_list.append('snap%d' % i)
+            self.snap_name_list.append(snap_name)
+
+            result_output = virsh.snapshot_list(self.vm.name,
+                                                ignore_status=False, debug=True)
+            if not utils_misc.wait_for(lambda: snap_name in result_output,
```

Comment:
I think checking if a string contains another string with 'wait_for' is meaningless, I think we dont need to check it several times, the results are same.

So, what about checking it as below:
```suggestion
            if not utils_misc.wait_for(lambda: snap_name in virsh.snapshot_list(self.vm.name,
                                                ignore_status=False, debug=True).stdout_text,
```

Reply:
Thank you Yingshun, get your point and updated